Frameworks/CoreText/CTLine.mm

```diff
@@ -370,15 +374,22 @@ CGFloat CTLineGetOffsetForStringIndex(CTLineRef lineRef, CFIndex charIndex, CGFl
                                 run->_stringIndices.begin() - 1;
 
                     if (index >= 0) {
-                        ret = std::accumulate(run->_dwriteGlyphRun.glyphAdvances, run->_dwriteGlyphRun.glyphAdvances + index, ret);
+                        if (_GlyphRunIsRTL(run->_dwriteGlyphRun)) {
+                            ret += std::accumulate(run->_dwriteGlyphRun.glyphAdvances,
+                                                   run->_dwriteGlyphRun.glyphAdvances + index,
+                                                   run->_relativeXOffset,
+                                                   std::minus<CGFloat>());
+                        } else {
+                            ret += std::accumulate(run->_dwriteGlyphRun.glyphAdvances,
+                                                   run->_dwriteGlyphRun.glyphAdvances + index,
+                                                   run->_relativeXOffset);
+                        }
                     }
```

Comment:
I tried that some other place but it _really_ didn't like it, can't remember why though.